src/main/java/com/simibubi/create/AllTileEntities.java

```diff
@@ -781,9 +785,9 @@ public class AllTileEntities {
 		.tileEntity("track", TrackTileEntity::new)
 		.instance(() -> TrackInstance::new)
 		.renderer(() -> TrackRenderer::new)
-		.validBlocks(AllBlocks.TRACK)
+		.validBlocks(((BlockEntry<? extends Block>[]) TrackMaterial.allBlocks().toArray(new BlockEntry[0])))
```

Comment:
mods extending TrackTileEntity / mixin to add to array / reflection should solve this, marking as resolved

Reply:
On second thought, extending the BE is too clunky. The alternative of mixin/reflection should not be needed in order to use an API designed to avoid those. The material could extend the array when created.